internal/resources/business_unit/resource.go

```diff
@@ -0,0 +1,934 @@
+						},
+						"first_name": schema.StringAttribute{
+							Description: "First name of the contact",
+							Optional:    true,
+						},
+						"last_name": schema.StringAttribute{
+							Description: "Last name of the contact",
+							Optional:    true,
+						},
+						"street_name": schema.StringAttribute{
+							Description: "Name of the street",
+							Optional:    true,
+						},
+						"street_number": schema.StringAttribute{
+							Description: "Street number",
+							Optional:    true,
+						},
+						"additional_street_info": schema.StringAttribute{
+							Description: "Further information on the street address",
+							Optional:    true,
+						},
+						"postal_code": schema.StringAttribute{
+							Description: "Postal code",
+							Optional:    true,
+						},
+						"city": schema.StringAttribute{
+							Description: "Name of the city",
+							Optional:    true,
+						},
+						"region": schema.StringAttribute{
+							Description: "Name of the region",
+							Optional:    true,
+						},
+						"state": schema.StringAttribute{
+							Description: "Name of the state",
+							Optional:    true,
+						},
+						"company": schema.StringAttribute{
+							Description: "Name of the company",
+							Optional:    true,
+						},
+						"department": schema.StringAttribute{
+							Description: "Name of the department",
+							Optional:    true,
+						},
+						"building": schema.StringAttribute{
+							Description: "Name or number of the building",
+							Optional:    true,
+						},
+						"apartment": schema.StringAttribute{
+							Description: "Name or number of the apartment",
+							Optional:    true,
+						},
+						"po_box": schema.StringAttribute{
+							Description: "Post office box number",
+							Optional:    true,
+						},
+						"phone": schema.StringAttribute{
+							Description: "Phone number",
+							Optional:    true,
+						},
+						"mobile": schema.StringAttribute{
+							Description: "Mobile phone number",
+							Optional:    true,
+						},
+						"email": schema.StringAttribute{
+							Description: "Email address",
+							Optional:    true,
+						},
+						"fax": schema.StringAttribute{
+							Description: "Fax number",
+							Optional:    true,
+						},
+						"additional_address_info": schema.StringAttribute{
+							Description: "Further information on the Address",
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"parent_unit": schema.ListNestedBlock{
+				Description: "Reference to a parent Business Unit by its key.",
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+					listvalidator.SizeAtLeast(1),
+					listvalidator.IsRequired(),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"key": schema.StringAttribute{
+							Description: "User-defined unique identifier of the Business Unit",
+							Required:    true,
+						},
+						"type_id": schema.StringAttribute{
```

Comment:
I think this one is always `business-unit`, so this can be removed (and the validation is incorrect). Instead we should add id as an optional input, and make key optional also